mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/ContextController.java

```diff
@@ -92,6 +94,30 @@ private JSONObject getContextInfo(File f) throws Exception {
                 }
             }
         }
+        
+        File rolePath = new File(pathCtx);
+        Collection<File> files = FileUtils.listFiles(rolePath, new String[] { "XML", "xml" }, false);
+        for (File curRoleFile : files) {
+            if (FilenameUtils.getBaseName(curRoleFile.toString()).equalsIgnoreCase(title)) {
+                DocumentBuilderFactory domFactory = DocumentBuilderFactory.newInstance();
+                DocumentBuilder builder = domFactory.newDocumentBuilder();
+                Document doc = builder.parse(curRoleFile);
+                XPath xpath = XPathFactory.newInstance().newXPath();
+                
+                // Parsing roles
+                XPathExpression xpRole = xpath.compile("//AllowedRoles/Role");
```

Comment:
you could use `//AlowedRoles/Role/text()` as xpath so that no need to check if it is a nodeset, it would allow you to get the actual value for the role more rapidly.